### DISCUSSION TITLE
Head only rendered client-side

### DISCUSSION THREAD

**Discussion by eivindml** [vercel/next.js#16901] [Help]
Hi,
I'm having issues with properly rendering meta tags using the `Head` (`next/head`) component. It only renders client-side, even though it's put inside `_app.tsx`, and I have removed all `getStaticProps()` calls etc.

Here is a sample code. 

```javascript
import { useState } from 'react'
import { NextPage, NextPageContext } from 'next'

interface Tickets {
  id: number
  type: string
  url?: string
}

interface AppProps {
  Component: any
  pageProps: any
  initialData: any
  allowCookies: string
}

const App = (
  title: string,
  url: string,
  store: any,
  tickets: Tickets
): NextPage<AppProps> => (props) => {
  const [showCookieNotice, setShowCookieNotice] = useState(
    props.allowCookies === undefined
  )

  const handleCookieSelect = () => {
    setShowCookieNotice(false)
  }

  return (
    <>
      <Head>
        <title>Test</title>
      </Head>
      {props.allowCookies == 'allow' && (
        <noscript
          dangerouslySetInnerHTML={{
            __html: `<iframe src="https://www.googletagmanager.com/ns.html?id=GTM-KSK97S" height="0" width="0" style="display:none;visibility:hidden;"></iframe>`
          }}
        />
      )}
      <props.Component {...props.pageProps} />
    </>
  )
}

export default App
```

If you are curious, the `App` is a curried function, since it's reused by multiple projects, like this:
```javascript
const App = AppCreator(title, url, store, tickets)

export default App
```

As you can see below, the title is not rendered from the server:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <style data-next-hide-fouc="true">
    body {
        display: none
    }
    </style>
    <noscript data-next-hide-fouc="true">
        <style>
        body {
            display: block
        }
        </style>
    </noscript>
    <link rel="icon" href="/favicon.ico"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    <link rel="preload" as="font" href="/fonts/Calibre-Medium.woff2" type="font/woff2" crossorigin="anonymous"/>
    <link rel="preload" as="font" href="/fonts/Calibre-Regular.woff2" type="font/woff2" crossorigin="anonymous"/>
    <link rel="preload" as="font" href="/fonts/Calibre-Light.woff2" type="font/woff2" crossorigin="anonymous"/>
    <link rel="preload" as="font" href="/fonts/Calibre-Thin.woff2" type="font/woff2" crossorigin="anonymous"/>
    <link rel="icon" href="/favicon.ico?v=2" type="image/x-icon"/>
    <link rel="shortcut icon" href="/favicon.ico?v=2" type="image/x-icon"/>
    <link rel="apple-touch-icon" href="/apple-touch-icon.png?v=2"/>
    <meta charSet="utf-8"/>
    <meta name="viewport" content="width=device-width"/>
    <meta name="next-head-count" content="2"/>
    <link rel="preload" href="/_next/static/development/pages/_app.js?ts=1599473969552" as="script"/>
    <link rel="preload" href="/_next/static/development/pages/index.js?ts=1599473969552" as="script"/>
    <link rel="preload" href="/_next/static/runtime/webpack.js?ts=1599473969552" as="script"/>
    <link rel="preload" href="/_next/static/runtime/main.js?ts=1599473969552" as="script"/>
    <noscript id="__next_css__DO_NOT_USE__"></noscript>
    <!-- removed css styles -->
</head>
<body>
    <div id="__next">
        <h1>Hellø wørld!</h1>
    </div>
    <script src="/_next/static/runtime/react-refresh.js?ts=1599473969552"></script>
    <script src="/_next/static/development/dll/dll_bf9f5a6409814f8e5869.js?ts=1599473969552"></script>
    <script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/","query":{},"buildId":"development","nextExport":true,"autoExport":true,"isFallback":false}</script>
    <script nomodule="" src="/_next/static/runtime/polyfills.js?ts=1599473969552"></script>
    <script async="" data-next-page="/_app" src="/_next/static/development/pages/_app.js?ts=1599473969552"></script>
    <script async="" data-next-page="/" src="/_next/static/development/pages/index.js?ts=1599473969552"></script>
    <script src="/_next/static/runtime/webpack.js?ts=1599473969552" async=""></script>
    <script src="/_next/static/runtime/main.js?ts=1599473969552" async=""></script>
    <script src="/_next/static/development/_buildManifest.js?ts=1599473969552" async=""></script>
    <script src="/_next/static/development/_ssgManifest.js?ts=1599473969552" async=""></script>
</body>
</html>
```

If I inspect the source code later, then I can see the the `<title>` has been added client-side.

**What can I do to render this server side?** It seems like som sites won't fetch social meta data when it's added client-side.

**Comment by rodilo** [comment]
I'm facing the same issue since upgrading to version 9.5.

**Comment by M1ck0** [comment]
Here is what I suggest you do. Create a new project with `create-next-app` and then add `<title>` there. If it works then you made a mistake somewhere (which we can then try to resolve). If not, then it's a NextJS bug and should be reported ASAP.

In this example you never imported
```js
import Head from "next/head"
```

which could be a reason why this is happening.

**Reply by rodilo** [reply]
Just checked with the previous version I was using, `9.4.0` it builds&exports ok, all meta tags are present in exported  HTML files.
Also, everything is imported correctly.

**Reply by lovettbarron** [reply]
@rodilo, did you find a solution for this? I'm running into the same issue and its proving to be a massive blocker. Head renders properly client side, but not server side so SEO tools don't see any of the meta tags.

**Reply by rodilo** [reply]
> @rodilo, did you find a solution for this? I'm running into the same issue and its proving to be a massive blocker. Head renders properly client side, but not server side so SEO tools don't see any of the meta tags.

Yeah, but the problem was that we have monorepo project, and one component library used a different version of `nextjs`. After I updated version to the same one I use in the main project all was good.

**Reply by lovettbarron** [reply]
> > @rodilo, did you find a solution for this? I'm running into the same issue and its proving to be a massive blocker. Head renders properly client side, but not server side so SEO tools don't see any of the meta tags.
> 
> Yeah, but the problem was that we have monorepo project, and one component library used a different version of `nextjs`. After I updated version to the same one I use in the main project all was good.

Ahh, interesting. Thank you. Time to do some digging then.

**Reply by lovettbarron** [reply]
Found the problem. I'd implemented the SSR side of styletron incorrectly.
